docs/spec/slashing/state.md

```diff
@@ -49,3 +50,31 @@ Where:
 * `IndexOffset` is incremented each time the candidate was a bonded validator in a block (and may have signed a precommit or not).
 * `JailedUntil` is set whenever the candidate is revoked due to downtime
 * `SignedBlocksCounter` is a counter kept to avoid unnecessary array reads. `SignedBlocksBitArray.Sum() == SignedBlocksCounter` always.
+
+### Slashing Period
+
+A slashing period is a start and end time associated with a particular validator,
+within which only the "worst infraction counts": the total amount of slashing for
+infractions committed within the period (and discovered whenever) is capped at the
+penalty for the worst offense.
+
+This period starts when a validator is first bonded and ends when a validator is slashed & jailed
+for double-signing (but does not end if they are slashed & jailed for just missing blocks).
+When the validator voluntarily unjails themselves (and possibly changes signing keys), they reset the period.
```

Comment:
Can you help clarify for me one case? What if a validator misses enough blocks to be slashed and jailed -- the slashing period doesn't end. The validator can then, after some time, unjail themselves, correct? If so, is the period reset even though it never ended (they never double signed)? If it is indeed reset, they only get penalized for liveliness cost?

I'm sure I'm missing something here, sorry.

Reply:
bump @cwgoes

Reply:
yup! you are right, this is unclear - addressing

Reply:
Actually, I don't think we need this, we should just end the period when they're unbonded for any reason.